src/elements/feedback.html

```diff
@@ -0,0 +1,289 @@
+          label="Leave a comment here (optional)"
+          no-label-float
+          value="{{comment}}"
+        ></paper-textarea>
+        <paper-button
+          primary
+          hidden$="[[!rating]]"
+          on-tap="_sendFeedback"
+          ga-on="click"
+          ga-event-category="feedback"
+          ga-event-action="send feedback"
+          ga-event-label$="submit the [[rating]] stars feedback"
+        >
+          [[buttonLabel]]
+        </paper-button>
+        <paper-button
+          class="delete-button"
+          hidden$="[[!showDeleteButton]]"
+          on-tap="_dispatchDeleteFeedback"
+          ga-on="click"
+          ga-event-category="feedback"
+          ga-event-action="delete feedback"
+          ga-event-label$="delete the feedback record"
+        >
+          {$ feedback.deleteFeedback $}
+        </paper-button>
+      </div>
+    </template>
+
+    <script>
+      class Feedback extends ReduxMixin(Polymer.Element) {
+        static get properties() {
+          return {
+            rating: {
+              type: Number,
+              value: 0,
+            },
+            comment: {
+              type: String,
+              value: '',
+            },
+            collection: {
+              type: String,
+              value: 'sessions',
+            },
+            dbItem: {
+              type: String,
+              observer: '_dbItemChanged',
+            },
+            user: {
+              type: Object,
+              statePath: 'user',
+              observer: '_userChanged',
+            },
+            previousFeedback: {
+              type: Object,
+              observer: '_previousFeedbackChanged',
+            },
+            feedbackFetching: {
+              type: Boolean,
+              statePath: 'feedback.fetching',
+            },
+            feedbackAdding: {
+              type: Boolean,
+              statePath: 'feedback.adding',
+              observer: '_feedbackAddingChanged',
+            },
+            feedbackAddingError: {
+              type: Object,
+              statePath: 'feedback.addingError',
+            },
+            feedbackDeleting: {
+              type: Boolean,
+              statePath: 'feedback.deleting',
+              observer: '_feedbackDeletingChanged',
+            },
+            feedbackDeletingError: {
+              type: Object,
+              statePath: 'feedback.deletingError',
+            },
+            buttonLabel: {
+              type: String,
+              value: 'Post',
+            },
+            showDeleteButton: {
+              type: Boolean,
+              value: false,
+            },
+          };
+        }
+
+        static get is() {
+          return 'feedback-block';
+        }
+
+        connectedCallback() {
+          super.connectedCallback();
+          // Listen for previousFeedback state changes
+          // eslint-disable-next-line no-undef
+          store.subscribe(() => this._updateFeedbackState());
```

Comment:
I this `store` usage (and the `_updateFeedbackState` use) will make #579 more difficult. Is there an easier way to do? Maybe with `statePath` somehow?

Reply:
I think I should re-do this PR some day, for Polymer 3.